File: ProblemSolve/120.cpp
/************
 * Triangle *
 ************/

class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        for(int row = triangle.size() - 2; row >= 0; --row){
            for(int col = 0; col < triangle[row].size(); ++col){
                triangle[row][col] += min(triangle[row + 1][col], triangle[row + 1][col + 1]);
            }
        }
        
        return triangle[0][0];
    }
};

File: ProblemSolve/120.cs
/************
 * Triangle *
 ************/

public class Solution {
    public int MinimumTotal(IList<IList<int>> triangle) {
        int row = 0;
        int col = 0;
        
        for(row = triangle.Count - 2; row >= 0; --row){
            for(col = 0; col < triangle[row].Count; ++col){
                triangle[row][col] += Math.Min(triangle[row+1][col], triangle[row+1][col+1]);
            }
        }
        
        return triangle[0][0];
    }
}